### PR TITLE
Redesign calendar frontend

### LINKED ISSUE #158: Redesign frontend
We introduced new styles for the calendar in the Learn WordPress redesign but these were kept as local overrides. We now need to update the Make WordPress calendar as well. There are designs for both, and we need to settle on styles and layout that cater to both:

### Designs

| Make | Learn - Calendar | Learn - List |
|--------|--------|-|
| Image | Image | Image |

### PR DESCRIPTION
**Pull request by adamwoodnz**
Closes #158 

Updates the layout and styles of the frontend to satisfy designs for [Learn WordPress Online Workshops](https://learn.wordpress.org/online-workshops/) and [Make WordPress meetings](https://make.wordpress.org/meetings/) calendars.

For the Make team colors used in the calendar, care has been taken to ensure contrast ratios between background and text meet WCAG 2.1.

A notable difference between the Make and Learn designs lies in the calendar/list toggle buttons. See screenshots on the [issue](#158). I've opted to use the icon buttons shown in the Learn design.

Lastly an RSS feed has been added to the feed links, which have moved from the filter bar to the bottom of the calendar view.

NOTE: Does not include fixing existing broken unit tests, or styling for past events mentioned in https://github.com/WordPress/meeting-calendar/pull/157. I think these should be done separately.

#### Make

| Desktop | Tablet | Mobile |
|-|-|-|
| make wordpress org_meetings__new-theme=1 | make wordpress org_meetings__new-theme=1(iPad) | make wordpress org_meetings__new-theme=1(Samsung Galaxy S20 Ultra) |

#### Learn

| Desktop | Tablet | Mobile |
|-|-|-|
| learn wordpress org_online-workshops_ | learn wordpress org_online-workshops_(iPad) | learn wordpress org_online-workshops_(Samsung Galaxy S20 Ultra) |

### Testing
Best tested in sandbox.
1. Load this branch of the plugin
2. Test with [Learn](https://learn.wordpress.org/online-workshops/) and [Make](https://make.wordpress.org/meetings/)
3. For Learn load [this branch](https://github.com/WordPress/Learn/pull/2945) as well
4. For Make test with both the old theme and the new theme using the theme switcher
5. Regression test all the things! Calendar view and event details modal, list view, team filter, feed links (GCal and ICS should be filtered when a team is selected), prev/next nav
6. Test responsiveness
7. Test a11y

### CODE REVIEW

**Comment by adamwoodnz** [comment]
@WordPress/meta-design your review would be appreciated too 🙏

**Comment by jasmussen** [comment]
Looks great, especially for a first version, contrasts pass and colors look good. 

Screenshot 2024-10-10 at 14 04 36

Potentially the cell padding, or the event margin (whichever is in play) can be reduced, so the events are closer together in density, but there's also some value in the separation. Can start here, then tweak if need be.

File: src/frontend/calendar/cell.js [hunk 1]
```diff
@@ -37,8 +37,8 @@ function CalendarCell( {
 			<strong>
 				<span className="screen-reader-text">
 					{ format( 'F j', date ) }{ ' ' }
-					{ // translators: %d: Count of all events, ie: 4.
-					sprintf(
+					{ sprintf(
+						// translators: %d: Count of all events, ie: 4.
```
Comment (adamwoodnz) [inline src/frontend/calendar/cell.js:41]: eslint caught these problems

File: src/frontend/calendar/index.js [hunk 2]
```diff
@@ -41,25 +47,33 @@ function Calendar() {
...
 					className="wporg-meeting-calendar__btn-group"
-					aria-label={ __( 'Month navigation', 'wporg-meeting-calendar' ) }
+					aria-label={ __(
+						'Month navigation',
+						'wporg-meeting-calendar'
+					) }
 				>
 					<Button
-						isSecondary
+						variant="secondary"
```
Comment (adamwoodnz) [inline src/frontend/calendar/index.js:56]: `isSecondary` is deprecated, `variant` is the new API, same goes for `isLink` and `isPrimary`

File: src/frontend/calendar/index.js [hunk 3]
```diff
@@ -69,35 +83,52 @@ function Calendar() {
...
 						} }
 						disabled={ shouldForceListView }
+						aria-label={ __(
+							'Calendar view',
+							'wporg-meeting-calendar'
+						) }
 					>
-						{ __( 'List', 'wporg-meeting-calendar' ) }
+						<CalendarIcon aria-hidden="true" focusable="false" />
 					</Button>
```
Comment (adamwoodnz) [inline src/frontend/calendar/index.js:132]: The order of these buttons has been swapped to fix the tab order.

File: src/frontend/filter/index.js [hunk 4]
```diff
@@ -101,38 +84,17 @@ const Filter = () => {
...
+						<span>
+							{ __(
+								'Remove team filter',
+								'wporg-meeting-calendar'
+							) }
+						</span>
 					</Button>
 				</>
 			) }
-			<div className="wporg-meeting-calendar__filter-feed">
```
Comment (adamwoodnz) [inline src/frontend/filter/index.js:109]: Extracted into Feed component and moved to bottom of UI

**Comment by adamwoodnz** [comment]
### Screenshots

#### Make list view

| Desktop | Tablet | Mobile |
|-|-|-|
| make wordpress org_meetings__new-theme=1(5 1440x900) (1) | make wordpress org_meetings__new-theme=1(iPad) (1) | make wordpress org_meetings__new-theme=1(Samsung Galaxy S20 Ultra) (1) |

#### Make with old theme

| Calendar | List |
|-|-|
| make wordpress org_meetings__new-theme=0(5 1440x900) | make wordpress org_meetings__new-theme=0(5 1440x900) (1) |

**Comment by jasmussen** [comment]
Massive step forward.

**Comment by fcoveram** [comment]
It looks great 👏 🚀

**Comment by ryelle** [comment]
Applied on my sandbox and make/meetings looks great with the new theme 🎉 

I found the old blue in a few places when previewing the new theme:

Screenshot 2024-10-11 at 5 41 19 PM
Screenshot 2024-10-11 at 5 42 57 PM
Screenshot 2024-10-11 at 6 01 31 PM

As far as a11y, I don't see any regressions. There are definitely things that could be improved, but they're current issues too. For example…

- As you tab through the calendar view, there's no indication of which day you're on. I'm not sure what best practice would be here, and a quick search isn't helping much.
- The list view day headings are not actually headings, so screen reader navigating through the page by heading levels also doesn't say which day things are.
- There are empty links in the list view, since all items are wrapped in `<a href={ event.link }>` (not just an a11y issue)

But those could be future improvements, not blockers for this style update.

**Comment by adamwoodnz** [comment]
> I found the old blue in a few places when previewing the new theme:

This'll be because I haven't shipped the components color vars in the new theme yet.

**Comment by adamwoodnz** [comment]
> * The list view day headings are not actually headings, so screen reader navigating through the page by heading levels also doesn't say which day things are.

Fixed in [1e2bae1](https://github.com/WordPress/meeting-calendar/pull/159/commits/1e2bae1bf9c4390dd29a3208d8b8a32cb8191b4e)

**Comment by adamwoodnz** [comment]
> * There are empty links in the list view, since all items are wrapped in `<a href={ event.link }>` (not just an a11y issue)

Fixed in [e25bce0](https://github.com/WordPress/meeting-calendar/pull/159/commits/e25bce0c02a1453fcf115c57cbab69d8d91b6933)

**Review by adelalhkeem2 — APPROVED** [review]

File: src/frontend/calendar/index.js [hunk 5]
```diff
@@ -41,25 +47,33 @@ function Calendar() {
...
 				<nav
 					className="wporg-meeting-calendar__btn-group"
-					aria-label={ __( 'Month navigation', 'wporg-meeting-calendar' ) }
+					aria-label={ __(
+						'Month navigation',
+						'wporg-meeting-calendar'
+					) }
 				>
 					<Button
-						isSecondary
```
[the remark this answers is not in the document]
Comment (adelalhkeem2) [reply src/frontend/calendar/index.js:47]: adb shell sh /storage/emulated/0/Android/data/moe.shizuku.privileged.api/start.sh

**Pull request merged**